File: Solutions/Easy/2154. Keep Multiplying Found Values by Two/Solution.cs
public class Solution {
    public int FindFinalValue(int[] nums, int original) {
        HashSet<int> seen = new HashSet<int>();
        foreach(int num in nums){
            if(num > original) seen.Add(num);
            else if(num == original){
                do{
                    original *= 2;
                }while(seen.Contains(original));
            }
        }
        return original;
    }
}

File: Solutions/Easy/2154. Keep Multiplying Found Values by Two/Solution.py
class Solution:
    def findFinalValue(self, nums: List[int], original: int) -> int:
        seen = set()
        for num in nums:
            if(num > original):
                seen.add(num)
            elif(num == original):
                original *= 2
                while(original in seen):
                    original *= 2
        return original